### DISCUSSION TITLE
middleware case sensitive

### DISCUSSION THREAD

**Discussion by vishalgiasoft009** [spatie/laravel-permission#2357] [Q&A] [answered]
I have created a Role Field when i used
Route::group(['middleware' => ['auth','role:field']], function() {
});
If admin can change role name case case sensitive field to Field OR Field to field then our middleware case failed.
When role name used small character 'field' then error " 403 User does not have the right role"

Screenshot: 
https://prnt.sc/RFYObCl63yno
https://prnt.sc/guWWQVZ3mEvj

Please help how to solve this problem

**Comment by erikn69** [comment]
> If admin can change role name case case sensitive field to Field OR Field

You must forget cache after manually changes, read documentation about that

**Comment by drbyte** [comment, marked as answer]
If you've configured your database to use a Case-Sensitive character-set, then searches/lookups will be case-sensitive.

I recommend you not allow users/administrators to change the name of the roles, since your application code can't be guaranteed to match. 

Perhaps a better option would be to give your administrators a drop-down list of roles to choose from, instead of typing them in manually?

**Reply by erikn69** [reply]
>  I recommend you not allow users/administrators to change the name of the roles

It would be the most logical, for example if we had many `@hasrole('field')/$user->hasRole('field')` in blade, and an admin change name to `Field` blade templates stop matching

**Reply by drbyte** [reply]
Correct. 
This is also why it's BEST to code your application around **permissions** and only use roles as a dynamic admin tool for attaching groups of permissions to users.
eg: https://spatie.be/docs/laravel-permission/best-practices/roles-vs-permissions

**Comment by drbyte** [comment]
Another idea: you could change your middleware to list multiple forms/spellings of the role name:
```diff
-['auth','role:field']]
+['auth','role:field|Field']]
```
As Erikn69 noted, the problem with that is you would need to do the same in Blade and Policies and Controllers, everywhere.
